docs/src/man/basics.md

```diff
@@ -1510,32 +1510,97 @@ julia> german[Not(5), r"S"]
+
+#### `function`
+Here `function` is a function which operates on an entire data frame
+column passed as a vector.
+If you instead want to apply a function to each element in the column,
+then you can wrap your element-wise function in `ByRow` like
+`ByRow(my_elementwise_function)`.
+This will apply `my_elementwise_function` to every element in the column
+and then collect the results back into a vector.
+When multiple columns are selected by `source_column_selector`,
+the `function` will receive the columns as multiple positional arguments in the
+order they are selected like `f(column1, column2, column3)`.
+Slurping and splatting with `...` can be used to define a function which can
+accept any number of columns returned by `source_column_selector`.
+For example, the function `f(columns...) = max.(columns...)` will return a new column
+containing the largest element from each row for any number of input columns.
+
+#### `new_column_names`
+The results of the transformation will be placed into new columns
+with names defined by `new_column_names`.
+`new_column_names` may be a string, a symbol, a vector of strings,
+or a vector of symbols.
+In the `source_column_selector => function => new_column_names` transformation form,
+`new_column_names` may additionally be a renaming function which operates on a string
+or a vector of strings to create the destination column names programmatically.
+In the `source_column_selector => function` transformation form,
+`new_column_names` will be generated automatically as the function name appended
+to the source column name with an underscore.
+However, if keyword argument `renamecols=false` is passed
+to the transformation function, then the new columns will
+retain their original source names instead of using automatically
+generated names.
+
+#### More Information
+This transformation pair syntax is sometimes referred to as a mini-language.
+More details and examples of the transformation mini-language can be found in
+[this blog post](https://bkamins.github.io/julialang/2020/12/24/minilanguage.html).
+
+!!! Note Notes
+    - Multiple transformations may be applied at once using any of the following:
+      - `transformation_function(source_dataframe, transformation1, transformation2)`
+      - `transformation_function(source_dataframe, [transformation1, transformation2])`
+      - `transformation_function(source_dataframe, [transformation1 transformation2])`
+    - Any of the `transformation` pair syntax forms shown above can also use
+      broadcasting with `.=>` to apply multiple transformations at once.
+      See the next section for examples.
+    - More complex transformations are available through the syntax
+      `transformation_function(function, source_dataframe)`.
+      This usage is more advanced,
+      so it is discussed in a separate section
+      [here](**TODO: Add a link to relevant info.**).
```

Comment:
I couldn't find a good place to link to for the transformation_function(function, source_dataframe) form. Even the API documentation doesn't seem to cover it much. I do not want to cover it in this section either. Maybe some external blog post exists?

Reply:
`transformation_function(function, source_dataframe)` is the same as `transformation_function(source_dataframe, function)` - there is nothing special to it except that it allows `do` syntax (and that is why `subset` and `subset!` do not support it)

Reply:
Oh, okay.